### DISCUSSION TITLE
Specify string length on Blueprint::morph method

### DISCUSSION THREAD

**Discussion by AntoineDewaele** [laravel/framework#43001] [Ideas]
Hi there !

I realize that the `morphs` method always setup a `string(255)` database column for the `{{morph}}_type` :

```php
public function numericMorphs($name, $indexName = null)
{
    $this->string("{$name}_type");

    $this->unsignedBigInteger("{$name}_id");

    $this->index(["{$name}_type", "{$name}_id"], $indexName);
}
``` 
But for some performances reasons, it could be usefull to reduce this lenght and the only way to do this right now is to manually create the morph type, morph id and indexes.

Maybe, we could add a third optionnal parameter to make this work, like for example :

```php
public function numericMorphs($name, $indexName = null, $morphTypeLength = null)
{
    $this->string("{$name}_type", $morphTypeLength);

    $this->unsignedBigInteger("{$name}_id");

    $this->index(["{$name}_type", "{$name}_id"], $indexName);
}
```

This way, we could use the method like this :

```php
Schema::table('reviews', function(Blueprint $table) {
    $table->morphs('author', null, 32);
});
```

I'm not sure about this parameter name but that could be the idea.
This change could be applied on all the morph methods like `nullableMorphs`, `nullableNumericMorphs`, `uuidMorphs` and so on..

What do you think aout this ?

**Comment by cdcvince** [comment]
I totally agree. Since the type is usually less than 10 characters, that's a pity to allocate this amount of memory needed in temporary table during complex query.

**Comment by cprobola** [comment]
Would be awesome !
